packages/expo-web-browser/android/src/test/java/expo/modules/webbrowser/WebBrowserModuleTest.kt

```diff
@@ -0,0 +1,277 @@
+    // then
+    promiseRejected(promise) {
+      assertFalse(it.rejectCodeSet)
+      assertFalse(it.rejectMessageSet)
+      assertTrue(it.rejectThrowableSet)
+      assertTrue(it.rejectThrowable is PackageManagerNotFoundException)
+    }
+  }
+
+  @Test
+  fun testArgumentsCorrectlyPassedToIntent() {
+    // given
+    val intentSlot = slot<Intent>()
+    val mock = mockCustomTabsActivitiesHelper(defaultCanResolveIntent = true, startIntentSlot = intentSlot)
+    initialize(mock)
+
+    // when
+    module.openBrowserAsync("http://expo.io", browserArguments(
+      toolbarColor = "#000000",
+      browserPackage = "com.browser.package",
+      enableBarCollapsing = true,
+      enableDefaultShareMenuItem = true,
+      showInRecents = true,
+      showTitle = true
+    ), promise)
+
+    intentSlot.captured.let {
+      assertEquals("com.browser.package", it.`package`)
+    }
+  }
+
+  @Test
+  fun testTrueFlagsCorrectlyPassedToIntent() {
+    // given
+    val intentSlot = slot<Intent>()
+    val mock = mockCustomTabsActivitiesHelper(defaultCanResolveIntent = true, startIntentSlot = intentSlot)
+    initialize(mock)
+
+    // when
+    module.openBrowserAsync("http://expo.io", browserArguments(
+      toolbarColor = "#000000",
+      browserPackage = "com.browser.package",
+      enableBarCollapsing = true,
+      enableDefaultShareMenuItem = true,
+      showInRecents = true,
+      showTitle = true
+    ), promise)
+
+    intentSlot.captured.let {
+      assertTrue(it.hasExtra(CustomTabsIntent.EXTRA_ENABLE_URLBAR_HIDING))
+      assertTrue(it.getBooleanExtra(CustomTabsIntent.EXTRA_ENABLE_URLBAR_HIDING, false))
+      assertTrue((it.flags and Intent.FLAG_ACTIVITY_NEW_TASK) > 0)
+      assertFalse((it.flags and Intent.FLAG_ACTIVITY_EXCLUDE_FROM_RECENTS) > 0)
+      assertFalse((it.flags and Intent.FLAG_ACTIVITY_NO_HISTORY) > 0)
+    }
+  }
+
+  @Test
+  fun testFalseFlagsCorrectlyPassedToIntent() {
+    // given
+    val intentSlot = slot<Intent>()
+    val mock = mockCustomTabsActivitiesHelper(defaultCanResolveIntent = true, startIntentSlot = intentSlot)
+    initialize(mock)
+
+    // when
+    module.openBrowserAsync("http://expo.io", browserArguments(
+      toolbarColor = "#000000",
+      browserPackage = "com.browser.package",
+      enableBarCollapsing = false,
+      enableDefaultShareMenuItem = false,
+      showInRecents = false,
+      showTitle = false
+    ), promise)
+
+    intentSlot.captured.let {
+      assertFalse(it.getBooleanExtra(CustomTabsIntent.EXTRA_ENABLE_URLBAR_HIDING, true))
+      assertTrue((it.flags and Intent.FLAG_ACTIVITY_NEW_TASK) > 0)
+      assertTrue((it.flags and Intent.FLAG_ACTIVITY_EXCLUDE_FROM_RECENTS) > 0)
+      assertTrue((it.flags and Intent.FLAG_ACTIVITY_NO_HISTORY) > 0)
+    }
+  }
+
+  @Test
+  fun testWarmUpWithGivenPackage() {
+  }
+
+  @Test
+  fun testWarmUpWithoutPackage() {
+
+  }
+
+  @Test
+  fun testCoolDownWithGivenPackage() {
+  }
+
+  @Test
+  fun testCoolDownWithoutPackage() {
+  }
```

Comment:
🤔 I guess it should be marked a little bit better that these tests are to written?

Reply:
It is already implemented, forgot to include it.